test/integration/play/test_play.py

```diff
@@ -2,13 +2,27 @@
 import re
 import shutil
 
-from utils import create_test_instance, run_command_and_get_output
+import pytest
+from utils import (create_test_instance, kill_instances,
+                   run_command_and_get_output)
 
 # The name of instance config file within this integration tests.
 # This file should be in /test/integration/play/test_file/.
 INSTANCE_NAME = "remote_instance_cfg.lua"
 
 
+# In case of unsuccessful completion of tests, tarantool test instances may remain running.
+# This is autorun wrapper for each test case in this module.
+@pytest.fixture(autouse=True)
+def kill_remain_instance_wrapper():
+    # Kill remain test instance if it is still running from past integration test.
+    kill_instances()
+    # Run test.
+    yield
+    # Kill a test instance if it was not stopped due to a failed test.
+    kill_instances()
```

Comment:
AFAIU, the second `kill_instances()` makes the first one no-op. So, maybe, it would be better to just assert that there are no 'orphan' instances?

Otherwise it looks like:

```python
if x:
    if x:  # For extra sure.
        foo()
```

Reply:
Done, now it works only before the `yield `